lib/blockscore/base.rb

```diff
@@ -45,16 +59,14 @@ def self.api_url
     end
 
     def self.endpoint
-      if self == Base
-        fail NotImplementedError, 'Base is an abstract class, not an API resource'
-      end
+      fail NotImplementedError, 'Base is an abstract class, not an API resource' if self == Base
```

Comment:
We could change the conditional to just `if equal?(Base)` which is cleaner and more strict